modules/@apostrophecms/ui/ui/apos/components/AposCellLabels.vue

```diff
@@ -3,7 +3,7 @@
     class="apos-table__cell-field apos-table__cell-field--labels"
     :class="`apos-table__cell-field--${header.name}`"
   >
-    <span v-if="manuallyPublished && item.modified && !item.submitted && item.lastPublishedAt">
+    <span v-if="manuallyPublished && (item.modified || draft.modified) && !item.submitted && item.lastPublishedAt">
```

Comment:
Use `draft && draft.modified`. `item` is for display purposes and isn't guaranteed to know about draft/live matters, it exists so you don't have to worry about that if all you want to do is display "what we think you should display."